bin/cylc

```diff
@@ -14,643 +15,225 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
+CATEGORIES=('control' 'con' 'information' 'info' 'all' 'task' 'license' \
+            'GPL' 'admin' 'prep' 'preparation' 'hook' 'discovery' 'utility' \
+            'util')
+                
+HELP_OPTS=('help' '--help' '-h' 'h' '?')
+
+get_version() {
+    # If there is a version file present, use that
+    if [[ -f "${CYLC_HOME}/VERSION" ]]; then
+        CYLC_VERSION="$(cat ${CYLC_HOME}/VERSION)"
+    else
+        local DESC=
+        DESC="$(cd ${CYLC_HOME} && git describe --abbrev=4 2>/dev/null)"
+        CYLC_VERSION="${DESC}"
+    fi
+    # Otherwise, use git to get the version information.
+    # Append "-dirty" if there are uncommitted changes.
+    if [[ -n "$(cd ${CYLC_HOME} && git status \
+            --untracked-files=no --porcelain)" ]]; then
+        CYLC_VERSION="${CYLC_VERSION}-dirty"
+    fi
+}
+
+print_version() {
+    get_version
+    if [[ "$#" -eq 0 ]]; then
+        echo "$CYLC_VERSION"
+    fi
+    if [[ "$@" == 'long' ]]; then
+        echo "Cylc ${CYLC_VERSION} (${CYLC_HOME})"
+    fi
+}
+
+init_cylc() {
+    set -eu
+    CYLC_HOME_BIN="$(dirname "$(readlink -f "$0")")"
+    CYLC_HOME="$(dirname "${CYLC_HOME_BIN}")"
+    CYLC_NS="$(basename $0)"
+    CYLC_DIR="${CYLC_HOME}"
+    
+    get_version
+    
+    PATH="$(path_lead "${PATH:-}" "${CYLC_HOME_BIN}")"
+    PYTHONPATH="$(path_lead "${PYTHONPATH:-}" "${CYLC_HOME}/lib/")"
+    PYTHONUNBUFFERED='true'
+    
+    export PATH PYTHONPATH PYTHONUNBUFFERED CYLC_DIR CYLC_VERSION
+}
+
+help_util() {
+    # Deals with form 'cylc [COMMAND] --help'
+    cd "${CYLC_HOME_BIN}"
+   # deal with graph which is a weird edge case...
+    if [[ "$@" == "graph" ]]; then
+        local COMMAND="${CYLC_HOME_BIN}/cylc-graph"
+        exec "${COMMAND}" "--help"
+        exit
```

Comment:
This `exec` statement above should replace the current process with the executed command, so this `exit` statement will never be reached.